<launch>
    <!-- number of moving objects -->
    <arg name="obj_num" value="10" />
    <arg name="drone_id" value="0"/>

    <arg name="map_size_x" value="50"/>
    <arg name="map_size_y" value="20"/>
    <arg name="map_size_z" value="3.0"/>
    <arg name="odom_topic" value="/mavros/local_position/odom"/>
    
    <!-- main algorithm params -->
    <include file="$(find ego_planner)/launch/advanced_param_exp.xml">
        <arg name="drone_id" value="$(arg drone_id)"/>
        <arg name="map_size_x_" value="$(arg map_size_x)"/>
        <arg name="map_size_y_" value="$(arg map_size_y)"/>
        <arg name="map_size_z_" value="$(arg map_size_z)"/>
        <arg name="odometry_topic" value="$(arg odom_topic)"/>
        <arg name="obj_num_set" value="$(arg obj_num)" />
        <!-- camera pose: transform of camera frame in the world frame -->
        <!-- depth topic: depth image, 640x480 by default -->
        <!-- don't set cloud_topic if you already set these ones! -->
        <arg name="camera_pose_topic" value="nouse1"/>
        <arg name="depth_topic" value="/camera/depth/image_rect_raw"/>
        <!-- topic of point cloud measurement, such as from LIDAR  -->
        <!-- don't set camera pose and depth, if you already set this one! -->
        <arg name="cloud_topic" value="nouse2"/>
        <!-- intrinsic params of the depth camera -->
        <arg name="cx" value="316.5154724121094"/>
        <arg name="cy" value="234.0104522705078"/>
        <arg name="fx" value="389.90655517578125"/>
        <arg name="fy" value="389.90655517578125"/>
        <!-- maximum velocity and acceleration the drone will reach -->
        <arg name="max_vel" value="1.30" />
        <arg name="max_acc" value="1.00" />
        <!--always set to 1.5 times grater than sensing horizen-->
        <arg name="planning_horizon" value="4.5" />
        <arg name="use_distinctive_trajs" value="false" />
        <!-- 1: use 2D Nav Goal to select goal  -->
        <!-- 2: use global waypoints below  -->
        <arg name="flight_type" value="2" />
        <!-- global waypoints -->
        <!-- It generates a piecewise min-snap traj passing all waypoints -->
        <arg name="point_num" value="4" />
        <arg name="point0_x" value="2.0" />
        <arg name="point0_y" value="0.0" />
        <arg name="point0_z" value="1.5" />

        <arg name="point1_x" value="3.0" />
        <arg name="point1_y" value="0.0" />
        <arg name="point1_z" value="1.5" />

        <arg name="point2_x" value="6.0" />
        <arg name="point2_y" value="1.0" />
        <arg name="point2_z" value="1.5" />

        <arg name="point3_x" value="9.0" />
        <arg name="point3_y" value="3.0" />
        <arg name="point3_z" value="1.80" />

        <arg name="point4_x" value="7.0" />
        <arg name="point4_y" value="2.5" />
        <arg name="point4_z" value="1.5" />

        <arg name="point5_x" value="4.0" />
        <arg name="point5_y" value="2.5" />
        <arg name="point5_z" value="1.5" />

        <arg name="point6_x" value="1.0" />
        <arg name="point6_y" value="1.0" />
        <arg name="point6_z" value="1.7" />  
    </include>

    <!-- trajectory server -->
    <node pkg="ego_planner" name="drone_$(arg drone_id)_traj_server" type="traj_server" output="screen">
        <param name="traj_server/time_forward" value="1.0" type="double"/>
    </node>     
    <!-- <include file="$(find uav_control)/launch/control.launch">
        <arg name="auto_offboard" value="False"/>
    </include> -->
</launch>
